### PR TITLE
feat: Artillery executor for testkube

### PR DESCRIPTION
**Pull request by abhishek9686**
## Checklist (choose what happened)

- [ ] breaking change! (describe)
- [x] tested locally
- [x] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [x] added a test

## Breaking changes

-

## Changes

Added changes for Artillery testkube executor

## Fixes

-

### CODE REVIEW

File: pkg/runner/artillery_helper.go [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+		return fmt.Errorf("can't find any content to run in execution data: %+v", execution)
+	}
+
+	if execution.Content.Repository == nil {
+		return fmt.Errorf("artillery executor handle only repository based tests, but repository is nil")
+	}
+
+	if execution.Content.Repository.Path == "" {
+		return fmt.Errorf("can't find repository path in params, repo:%+v", execution.Content.Repository)
+	}
```
Comment (exu) [inline pkg/runner/artillery_helper.go:160]: This one should be removed - just noticed that we have an issue in template (was fixing Cypress issue recently)

File: Makefile [hunk 2]
```diff
@@ -1,4 +1,4 @@
-NAME ?= testkube-executor-template
+NAME ?= testkube-artillery-executor
```
[the remark this answers is not in the document]
Comment (abhishek9686) [reply Makefile]: sure I have made the change

File: pkg/runner/artillery_helper.go (hunk 1 above)
Comment (abhishek9686) [reply pkg/runner/artillery_helper.go:160]: But isn't this check necessary for git-based executors to check if the  test files path is not mentioned for the given git repo

Comment (exu) [reply pkg/runner/artillery_helper.go:160]: Yeah, but we also are handling `""` as root currently, I think it was some leftover in template :/

Comment (abhishek9686) [reply pkg/runner/artillery_helper.go]: oh alright, I have removed the check

File: pkg/runner/artillery_helper.go [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+	Median float64 `json:"median"`
+	P75    float64 `json:"p75"`
+	P90    float64 `json:"p90"`
+	P95    float64 `json:"p95"`
+	P99    float64 `json:"p99"`
+	P999   float64 `json:"p999"`
+}
+
+// ArtilleryTestResult ...
+type ArtilleryTestResult struct {
```
[the remark this answers is not in the document]
Comment (abhishek9686) [reply pkg/runner/artillery_helper.go:102]: done

**Review by exu — APPROVED** [review]
LGTM 🎉

**Comment by exu** [comment]
@abhishek9686 tests are failing, rest looks ok 👍🏼

**Comment by abhishek9686** [comment]
> @abhishek9686 tests are failing, rest looks ok 👍🏼

okay i will fix it

**Comment by abhishek9686** [comment]
> @abhishek9686 tests are failing, rest looks ok 👍🏼

@exu  tesst should pass once you merge the changes as artillery test files are being pulled from main branch since the changes are not in the main branch tests are failing

**Pull request merged**